lib/location/method_cloud_location.c

```diff
@@ -149,21 +135,18 @@ static void method_cloud_location_positioning_work_fn(struct k_work *work)
 		location_result.latitude = location.latitude;
 		location_result.longitude = location.longitude;
 		location_result.accuracy = location.accuracy;
-		if (running) {
-			running = false;
-			location_core_event_cb(&location_result);
-		}
+		location_core_event_cb(&location_result);
```

Comment:
Just a thought and it's up to you if you do it or not, but this could be moved to the end of the function, where `location_core_event_cb_timeout()` and `location_core_event_cb_error()` are called. Then all event callback calls would be in one place, like clearing of the `running` flag is now.

Reply:
`location_core_event_cb()` shouldn't be sent when running is checked on line 72.
Also, it's kind of nice to call this right after the `location_result` is set on lines 135-137.

Reply:
> `location_core_event_cb()` shouldn't be sent when running is checked on line 72.

Ok, I didn't spot that because it wasn't visible in the diff by default.